ros/src/visualization/vis_node.py

```diff
@@ -87,6 +91,18 @@ def handle_traffic_lights_msg(self, traffic_lights_msg):
     def handle_traffic_waypoint_msg(self, traffic_waypoint_msg):
         self.traffic_waypoint = traffic_waypoint_msg.data
 
+    # Callback to adjust dynamic variables
+    def dyn_vars_cb(self, config, level):
+
+        if self.vis_rate != config['dyn_vis_rate']:
+            self.vis_rate = config['dyn_vis_rate']  # store new rate
+            self.rate = rospy.Rate(self.vis_rate)  # apply new rate
+
+        if self.vis_enabled != config['dyn_vis_enabled']:
```

Comment:
I think you could just assign the value - no need to check first here.

Reply:
Yea, you're right.  This callback only executes when something was updated so might as well just set the param so I updated this.  I left it on the rospy.Rate() call though just in case calling even with the same rate value resets some of the timers so I don't want to call it unnecessarily.